packaging/pulp_repo.py

```diff
@@ -0,0 +1,644 @@
+            if value is not None:
+                importer_config[key] = value
+
+        data['importer_config'] = importer_config
+
+        if self.repo_type == 'rpm':
+            data['importer_type_id'] = "yum_importer"
+
+        response, info = fetch_url(
+            self.module,
+            url,
+            data=json.dumps(data),
+            method='POST')
+
+        if info['status'] != 202:
+            self.module.fail_json(
+                msg="Failed to set the repo importer configuration",
+                status_code=info['status'],
+                response=info['msg'],
+                importer_config=importer_config,
+                url=url)
+
+    def set_repo_list(self):
+        url = "%s/pulp/api/v2/repositories/?details=true" % self.host
+        response, info = fetch_url(self.module, url, method='GET')
+
+        if info['status'] != 200:
+            self.module.fail_json(
+                msg="Request failed",
+                status_code=info['status'],
+                response=info['msg'],
+                url=url)
+
+        self.repo_list = json.load(response)
+
+    def verify_tasks_completed(self, response_dict):
+        for task in response_dict['spawned_tasks']:
+            task_url = "%s%s" % (self.host, task['_href'])
+
+            while True:
+                response, info = fetch_url(
+                    self.module,
+                    task_url,
+                    data='',
+                    method='GET')
+
+                if info['status'] != 200:
+                    self.module.fail_json(
+                        msg="Failed to check async task status.",
+                        status_code=info['status'],
+                        response=info['msg'],
+                        url=url)
+
+                task_dict = json.load(response)
+
+                if task_dict['state'] == 'finished':
+                    return True
+
+                if task_dict['state'] == 'error':
+                    self.module.fail_json(msg="Asynchronous task failed to complete.", error=task_dict['error'])
+
+                sleep(2)
+
+
+def main():
+    argument_spec = url_argument_spec()
+    argument_spec.update(
+        add_export_distributor=dict(default=False, type='bool'),
+        feed_url=dict(),
+        importer_ssl_ca_cert=dict(),
+        importer_ssl_client_cert=dict(),
+        importer_ssl_client_key=dict(),
+        name=dict(required=True, aliases=['repo']),
+        proxy_host=dict(),
+        proxy_port=dict(),
+        publish_distributor=dict(),
+        pulp_host=dict(default="http://127.0.0.1"),
+        relative_url=dict(),
+        repo_type=dict(default="rpm"),
+        serve_http=dict(default=False, type='bool'),
+        serve_https=dict(default=True, type='bool'),
+        state=dict(
+            default="present",
+            choices=['absent', 'present', 'sync', 'publish']),
+        wait_for_completion=dict(type="bool", default=False))
```

Comment:
Please could you swap the `type` with the `default` parameters to have them in the same order as in the cases above?